<a href="https://colab.research.google.com/github/teoxami/titanic-data-analysis-python/blob/main/01_IntroPython__MaitaTeo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Análisis de datos del Titanic usando Python y Pandas

## 📂 Carga del Dataset
En esta sección se carga el archivo `titanic.csv` en el entorno de Google Colab para su posterior análisis.


In [ ]:
from google.colab import files
uploaded = files.upload()


Saving titanic.csv to titanic (2).csv


## 📊 Lectura del Dataset
Se carga el dataset en un DataFrame de pandas para poder manipular y analizar la información.


In [ ]:
# Importar librerías necesarias
import pandas as pd
import numpy as np

# Cargar el CSV a un DataFrame
df = pd.read_csv('titanic.csv')

# Ver las primeras filas para confirmar
print("Primeras 5 filas del dataset:")
df.head()

Primeras 5 filas del dataset:


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


## 📈 Análisis Estadístico General
Se utilizan estadísticas descriptivas para comprender la distribución y características principales de los datos.


In [ ]:
# Estadísticas descriptivas de los datos numéricos
print("Estadísticas descriptivas:")
df.describe()

Estadísticas descriptivas:


,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


## 🛟 Análisis de Supervivencia
En esta sección se calcula el porcentaje de pasajeros que sobrevivieron al accidente del Titanic.


In [ ]:
# BLOQUE 1: Análisis de supervivencia
print("="*50)
print("ANÁLISIS DE SUPERVIVENCIA")
print("="*50)

# Calcular porcentaje de supervivencia
supervivencia = df['Survived'].value_counts(normalize=True) * 100

print(f"✓ Porcentaje que SOBREVIVIÓ: {supervivencia[1]:.2f}%")
print(f"✗ Porcentaje que NO sobrevivió: {supervivencia[0]:.2f}%")
print(f"\n Total de pasajeros: {len(df)} personas")
print(f"   Sobrevivieron: {df['Survived'].sum()} personas")
print(f"   Fallecieron: {len(df) - df['Survived'].sum()} personas")

ANÁLISIS DE SUPERVIVENCIA
✓ Porcentaje que SOBREVIVIÓ: 38.38%
✗ Porcentaje que NO sobrevivió: 61.62%

 Total de pasajeros: 891 personas
   Sobrevivieron: 342 personas
   Fallecieron: 549 personas


## 🎟️ Análisis por Clase
Se analiza la distribución de los pasajeros según su clase (primera, segunda y tercera clase).


In [ ]:
# BLOQUE 2: Análisis de clase (Pclass)
print("\n" + "="*50)
print("ANÁLISIS POR CLASE (Pclass)")
print("="*50)

# Calcular porcentaje de pasajeros por cada clase
porcentaje_clase = df['Pclass'].value_counts(normalize=True) * 100
conteo_clase = df['Pclass'].value_counts()

# Mostrar resultados ordenados por clase
for clase in sorted(df['Pclass'].unique()):
    print(f"Clase {clase}: {porcentaje_clase[clase]:.2f}% del total ({conteo_clase[clase]} pasajeros)")

# Información adicional: supervivencia por clase
print("\n Supervivencia por clase:")
for clase in sorted(df['Pclass'].unique()):
    sobreviven_clase = df[df['Pclass'] == clase]['Survived'].mean() * 100
    print(f"   Clase {clase}: {sobreviven_clase:.2f}% de supervivencia")


ANÁLISIS POR CLASE (Pclass)
Clase 1: 24.24% del total (216 pasajeros)
Clase 2: 20.65% del total (184 pasajeros)
Clase 3: 55.11% del total (491 pasajeros)

 Supervivencia por clase:
   Clase 1: 62.96% de supervivencia
   Clase 2: 47.28% de supervivencia
   Clase 3: 24.24% de supervivencia


## 🚻 Análisis por Género
Se determina la cantidad de hombres y mujeres que viajaban en el Titanic.


In [ ]:
# BLOQUE 3: Análisis por género
print("\n" + "="*50)
print("ANÁLISIS POR GÉNERO")
print("="*50)

# Contar hombres y mujeres
conteo_genero = df['Sex'].value_counts()
porcentaje_genero = df['Sex'].value_counts(normalize=True) * 100

print(f" Hombres: {conteo_genero.get('male', 0)} ({porcentaje_genero.get('male', 0):.2f}%)")
print(f" Mujeres: {conteo_genero.get('female', 0)} ({porcentaje_genero.get('female', 0):.2f}%)")

# Información adicional: supervivencia por género
print("\n Supervivencia por género:")
mujeres_superviven = df[df['Sex'] == 'female']['Survived'].mean() * 100
hombres_superviven = df[df['Sex'] == 'male']['Survived'].mean() * 100
print(f"    Mujeres: {mujeres_superviven:.2f}% sobrevivieron")
print(f"    Hombres: {hombres_superviven:.2f}% sobrevivieron")


ANÁLISIS POR GÉNERO
 Hombres: 577 (64.76%)
 Mujeres: 314 (35.24%)

 Supervivencia por género:
    Mujeres: 74.20% sobrevivieron
    Hombres: 18.89% sobrevivieron


##  Mis conclusiones personales

Después de analizar los datos del Titanic, puedo concluir:

1. **Sobre la supervivencia**:
   - Solo el 38.38% de los pasajeros sobrevivió, lo que muestra que fue una tragedia con alta mortalidad.

2. **Sobre las clases sociales**:
   - La mayoría de los pasajeros viajaban en tercera clase, lo que refleja la composición socioeconómica del barco.

3. **Sobre el género**:
   - Viajaron más hombres que mujeres, pero sobrevivieron mas mujeres que hombres.

### Reflexión final
La información generada a partir de estos datos nos permite entender mejor los patrones de supervivencia y la composición de los pasajeros del Titanic. Esto demuestra cómo los datos crudos se convierten en información útil para tomar decisiones o comprender eventos históricos.